tests/test_plugin_detection.py

```diff
@@ -74,29 +73,46 @@ def setUp(self):
         self.entry_patcher.start()
 
     def tearDown(self):
-        self.module_patcher.stop()
+        self.sys_patch.stop()
         self.entry_patcher.stop()
+        del(sys.modules['otio_mockplugin'])
 
-    def test_detect_pugin(self):
+    def test_detect_plugin(self):
         # Create a manifest and ensure it detected the mock adapter and linker
         man = otio.plugins.manifest.load_manifest()
-        self.assertIn(self.mock_adapter, man.adapters)
-        self.assertIn(self.mock_linker, man.media_linkers)
-        self.assertNotIn(self.mock_linker, man.adapters)
-        self.assertNotIn(self.mock_adapter, man.media_linkers)
 
-    def test_failed_plugin_load(self):
-        # Disable the error logging to keep the test from being scary
-        logging.disable(logging.CRITICAL)
+        # Make sure the adapter is included in the adapter list
+        adapter_names = [adapter.name for adapter in man.adapters]
+        self.assertIn('mock_adapter', adapter_names)
 
-        self.mock_plugin_module.plugin_manifest = mock.Mock(
-            side_effect=Exception
-        )
+        # Make sure the linker is included in the linker list
+        linker_names = [linker.name for linker in man.media_linkers]
+        self.assertIn('mock_linker', linker_names)
 
-        # Ensure we can load the manifest, safely
+        # Make sure adapters and linkers landed in the proper place
+        for adapter in man.adapters:
+            self.assertTrue(isinstance(adapter, otio.adapters.Adapter))
```

Comment:
There is a `assertIsInstance` function,: 
https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertIsInstance